### PR TITLE
CPR-569: New command to delete all finished pods

### PR DESCRIPTION
**Pull request by barabasicsongor**
This PR is a step towards being able to have multiple runs for a pipeline concurrently in Jenkins. A new command is added to paddle, which has the goal to clean up all the pods that are in `Succeeded, Failed or Unknown` states.

The goal of this ticket is that we can have a pipeline in Jenkins, that is scheduled to run every day and execute this clean up command.

[CPR-569](https://deliveroo.atlassian.net/browse/CPR-569)

### CODE REVIEW

File: cli/root.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package cli
 
 import (
 	"fmt"
+	"github.com/deliveroo/paddle/cli/cleanup"
```
Comment (barabasicsongor) [inline cli/root.go:18]: Haven't run the linter yet, will fix this soon

File: cli/pipeline/template.go [hunk 2]
```diff
@@ -191,7 +192,8 @@ func NewPodDefinition(pipelineDefinition *PipelineDefinition, pipelineDefinition
 	stepName := sanitizeName(pipelineDefinitionStep.Step)
 	branchName := sanitizeName(pipelineDefinitionStep.Branch)
 	stepVersion := sanitizeName(pipelineDefinitionStep.Version)
-	podName := fmt.Sprintf("%s-%s-%s-%s", sanitizeName(pipelineDefinition.Pipeline), sanitizeName(pipelineDefinitionStep.Version), stepName, branchName)
+	timestamp := time.Now().UTC()
+	podName := fmt.Sprintf("%s-%s-%s-%s-%s", sanitizeName(pipelineDefinition.Pipeline), sanitizeName(pipelineDefinitionStep.Version), stepName, branchName, timestamp.String())
```
Comment (alexstoick) [inline cli/pipeline/template.go:196]: lets check the max length of pod names - to make sure we avoid running into those limits. 

I think if you just add the last 4 digits of the integer time - you'd very rarely run into collisions.

File: cli/cleanup/pods.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func runPodsCleanup() {
+	pods := clientset.CoreV1().Pods("")
+	podList, err := pods.List(metav1.ListOptions{})
+	if err != nil {
+		logFatalf("[paddle] error fetching list of pods: %s", err.Error())
+	}
+
+	for _, pod := range podList.Items {
+		switch string(pod.Status.Phase) {
+		case "Succeeded", "Failed", "Unknown":
```
Comment (alexstoick) [inline cli/cleanup/pods.go:50]: I've never seen `Unknown` state before. Also for `Failed` - i'd suggest you only delete those if they've been around for more than 1 hour - to allow people to inspect the logs.

File: cli/pipeline/template.go (hunk 2 above)
Comment (barabasicsongor) [reply cli/pipeline/template.go:196]: The limit is 253 characters. Will modify by having the last 4 digits then

**Pull request merged**